### PR TITLE
Use PreferenceKeys file to manage SharedPreferences keys.

### PR DESCRIPTION
**Pull request by Skrilltrax**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## :scroll: Description
Let's just stop using raw strings :)

## :bulb: Motivation and Context
Less chances of breaking code due to mistakes in typing strings again

## :green_heart: How did you test it?
Ran the app and changed a few settings. Still needs one more round of testing though.

## :pencil: Checklist

- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

## :crystal_ball: Next steps
Port viscerion's application preferences.

### CODE REVIEW

File: app/src/main/java/com/zeapo/pwdstore/utils/PreferenceKeys.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    const val REPO_CHANGED = "repo_changed"
+    const val EXPORT_PASSWORDS = "export_passwords"
+    const val ENABLE_DEBUG_LOGGING = "enable_debug_logging"
+    const val BIOMETRIC_AUTH = "biometric_auth"
+    const val PREF_KEY_CUSTOM_DICT = "pref_key_custom_dict"
+    const val PREF_KEY_IS_CUSTOM_DICT = "pref_key_is_custom_dict"
+    const val PREF_KEY_PWGEN_TYPE = "pref_key_pwgen_type"
+    const val OPENPGP_KEY_IDS_SET = "openpgp_key_ids_set"
+
+}
```
Comment (msfjarvis) [inline app/src/main/java/com/zeapo/pwdstore/utils/PreferenceKeys.kt]: needs a trailing newline, run a format pass with Android Studio.

**Review by msfjarvis — DISMISSED** [review]

File: app/src/main/java/com/zeapo/pwdstore/utils/PreferenceKeys.kt (hunk 1 above)
Comment (Skrilltrax) [reply app/src/main/java/com/zeapo/pwdstore/utils/PreferenceKeys.kt]: Cool, need to update a lot more files though.

**Comment by Skrilltrax** [comment]
I think I've updated all the references of the keys. I've updated the PreferenceKeys with `region` and `endregion`, do let me know if you don't like it in the codebase.

**Comment by msfjarvis** [comment]
> I think I've updated all the references of the keys. I've updated the PreferenceKeys with `region` and `endregion`, do let me know if you don't like it in the codebase.

I'm not a huge fan of the comments so I'd prefer reverting it. Also bring your branch up-to-date with `develop` since that's enforced by branch protection.

**Comment by msfjarvis** [comment]
Completely missed some files, and PasswordStore.kt has a `PREFERENCE_SEEN_AUTOFILL_ONBOARDING` field that needs to go into `PreferenceKeys` as well.

```
app/src/main/java/com/zeapo/pwdstore/Application.kt:        AppCompatDelegate.setDefaultNightMode(when (prefs?.getString("app_theme", getString(R.string.app_theme_def))) {
app/src/main/java/com/zeapo/pwdstore/crypto/DecryptActivity.kt:                            val showPassword = settings.getBoolean("show_password", true)
app/src/main/java/com/zeapo/pwdstore/crypto/DecryptActivity.kt:                            val showExtraContent = settings.getBoolean("show_extra_content", true)
app/src/main/java/com/zeapo/pwdstore/crypto/DecryptActivity.kt:                            if (settings.getBoolean("copy_on_decrypt", true)) {
app/src/main/java/com/zeapo/pwdstore/autofill/AutofillService.kt:        val defValue = if (settings!!.getBoolean(PreferenceKeys.AUTOFILL_DEFAULT, true)) "/first" else "/never"
app/src/main/java/com/zeapo/pwdstore/autofill/AutofillService.kt:        // if autofill_default is checked and prefs.getString DNE, 'Automatically match with password'/"first" otherwise "never"
app/src/main/java/com/zeapo/pwdstore/autofill/AutofillService.kt:        val defValue = if (settings!!.getBoolean("autofill_default", true)) "/first" else "/never"
app/src/main/java/com/zeapo/pwdstore/autofill/AutofillService.kt:                        val ttl = Integer.parseInt(settings!!.getString(PreferenceKeys.GENERAL_SHOW_TIME, "45")!!)
```

**Comment by Skrilltrax** [comment]
Oh thanks, I'll revisit all the files again. IntelliJ's search results aren't that good when you have 100+ matching items.

**Review by msfjarvis — DISMISSED** [review]
LGTM

**Review by fmeum — DISMISSED** [review]

**Review by fmeum — APPROVED** [review]

**Review by msfjarvis — APPROVED** [review]

**Pull request merged**